hollow/src/main/java/com/netflix/hollow/core/read/engine/HollowBlobReader.java

```diff
@@ -147,6 +165,17 @@ public void readSnapshot(HollowBlobInput in, TypeFilter filter) throws IOExcepti
             typeNames.add(typeName);
         }
 
+        if(optionalParts != null) {
+            for(Map.Entry<String, HollowBlobInput> optionalPartEntry : optionalParts.getInputsByPartName().entrySet()) {
+                numStates = VarInt.readVInt(optionalPartEntry.getValue());
+
+                for(int i=0;i<numStates;i++) {
+                    String typeName = readTypeStateSnapshot(optionalPartEntry.getValue(), filter);
```

Comment:
Just wanted to confirm, does the order in which `readTypeStateSnapshot` is called on types matter? I noticed we use a `TreeSet` above

Reply:
It shouldn't matter.  During an update references on some types may be incorrect *but* none of the consumer's indexing structures should point to any of the new data until the update is fully complete.